Using error in Contains fails with confusing message

https://play.golang.org/p/u7Dfo4UnxZ0
```go
assert.Contains(t, "error string", errors.New("error"))
```
```
Error Trace:	prog.go:25
Error:      	"error string" does not contain "error"
Test:       	TestCaseA
```

Whereas this is what's needed to pass:
```diff
-assert.Contains(t, "error string", errors.New("error"))
+assert.Contains(t, "error string", errors.New("error").Error())
```

I would prefer if the failure message explained that it was an error type, otherwise it looks like a string does not contain another string: I've fallen foul to this in the past, and just spent 5-10mins again not realising that I needed to pass a `string` type as the 3rd argument.

It would be really nice to have the type printed here when the value isn't a `Stringer` =)

I think the current behavior is correct, but I agree that the error message is misleading.

Here's the source of the problem:
https://github.com/stretchr/testify/blob/6be346c1f15898af82c7c86dbad99cc9d7fee6f2/assert/assertions.go#L765-L767

One solution might be to change the `%s` to a `%#v` like this:

```diff
-return Fail(t, fmt.Sprintf("\"%s\" does not contain \"%s\"", s, contains), msgAndArgs...)
+return Fail(t, fmt.Sprintf("%#v does not contain %#v", s, contains), msgAndArgs...)
```

But that might be overly verbose:

```
--- FAIL: TestCaseA (0.00s)
    test.go:25:
                Error Trace:    test.go:25
                Error:          "test" does not contain &errors.errorString{s:"test"}
                Test:           TestCaseA
```

Or maybe it's just the right amount of verbosity. I'm not sure.

--- a/assert/assertions.go
+++ b/assert/assertions.go
@@ -760,10 +760,10 @@ func Contains(t TestingT, s, contains interface{}, msgAndArgs ...interface{}) bo
 
 	ok, found := includeElement(s, contains)
 	if !ok {
-		return Fail(t, fmt.Sprintf("\"%s\" could not be applied builtin len()", s), msgAndArgs...)
+		return Fail(t, fmt.Sprintf("%#v could not be applied builtin len()", s), msgAndArgs...)
 	}
 	if !found {
-		return Fail(t, fmt.Sprintf("\"%s\" does not contain \"%s\"", s, contains), msgAndArgs...)
+		return Fail(t, fmt.Sprintf("%#v does not contain %#v", s, contains), msgAndArgs...)
 	}
 
 	return true

--- a/assert/assertions_test.go
+++ b/assert/assertions_test.go
@@ -592,6 +592,18 @@ func TestContains(t *testing.T) {
 	}
 }
 
+func TestContainsFailMessage(t *testing.T) {
+
+	mockT := new(mockTestingT)
+
+	Contains(mockT, "Hello World", errors.New("Hello"))
+	expectedFail := "\"Hello World\" does not contain &errors.errorString{s:\"Hello\"}"
+	actualFail := mockT.errorString()
+	if !strings.Contains(actualFail, expectedFail) {
+		t.Errorf("Contains failure should include %q but was %q", expectedFail, actualFail)
+	}
+}
+
 func TestNotContains(t *testing.T) {
 
 	mockT := new(testing.T)

